test/specs/rules/require-spaces-inside-parenthesized-expression.js

```diff
@@ -0,0 +1,169 @@
+            assert(checker.checkString('if (( 1 + 2 ))\n    3').isEmpty());
+            assert(checker.checkString('( my )(( a ),( b ))').isEmpty());
+        });
+    });
+
+    describe('exceptions', function() {
+        it('should not report for function or object when so configured', function() {
+            checker.configure({
+                requireSpacesInsideParenthesizedExpression: {
+                    allExcept: ['{', '}', 'function']
+                }
+            });
+
+            assert(checker.checkString('([])').getErrorCount() === 2);
+            assert(checker.checkString('("string")').getErrorCount() === 2);
+            assert(checker.checkString('(i)').getErrorCount() === 2);
+            assert(checker.checkString('(1)').getErrorCount() === 2);
+            assert(checker.checkString('(function() {}, {})').isEmpty());
+        });
+
+        it('should not report for object when so configured', function() {
+            checker.configure({
+                requireSpacesInsideParenthesizedExpression: {
+                    allExcept: ['{', '}']
+                }
+            });
+
+            assert(checker.checkString('([])').getErrorCount() === 2);
+            assert(checker.checkString('("string")').getErrorCount() === 2);
+            assert(checker.checkString('(i)').getErrorCount() === 2);
+            assert(checker.checkString('(1)').getErrorCount() === 2);
+            assert(checker.checkString('(function() {}, {})').getErrorCount() === 1);
+            assert(checker.checkString('( function() {}, {})').isEmpty());
+        });
+
+        it('should consider comments', function() {
+            checker.configure({
+                requireSpacesInsideParenthesizedExpression: {
+                    allExcept: ['//', '/*', '*/']
+                }
+            });
+
+            assert(checker.checkString('(foo)').getErrorCount() === 2);
+            assert(checker.checkString('(/**/ foo /**/)').isEmpty());
+            assert(checker.checkString('(//\n foo //\n)').isEmpty());
+            assert(checker.checkString('(/*comment*/ foo /*comment*/)').isEmpty());
+            assert(checker.checkString('(//comment\n foo //comment\n)').isEmpty());
+        });
+
+        it('should differentiate comments', function() {
+            checker.configure({
+                requireSpacesInsideParenthesizedExpression: {
+                    allExcept: ['/*', '*/']
+                }
+            });
+
+            assert(checker.checkString('(foo)').getErrorCount() === 2);
+            assert(checker.checkString('(/**/ foo /**/)').isEmpty());
+            assert(checker.checkString('(//\n foo //\n)').getErrorCount() === 1);
+            assert(checker.checkString('(/*comment*/ foo /*comment*/)').isEmpty());
+            assert(checker.checkString('(//comment\n foo //comment\n)').getErrorCount() === 1);
+        });
+
+        it('should not look inside comments', function() {
```

Comment:
Yes. These rules should look only at the characters beginning or ending comments, not at their contents.